port/unix/omrsysinfo.c

```diff
@@ -6131,71 +6146,68 @@ populateCgroupEntryListV2(struct OMRPortLibrary *portLibrary, int pid, OMRCgroup
 					osErrCode);
 			goto _end;
 		}
-	} else {
-		/* For v2, the cgroup file contains one line of the format '0::<cgroup>'. */
-		char cgroup[PATH_MAX];
-		/* This array should be large enough to read names of all subsystems. 1024 should be enough based on current supported subsystems. */
-		char subsystems[PATH_MAX];
-		char *cursor = NULL;
-		char *separator = NULL;
 
 		rc = sscanf(buffer, PROC_PID_CGROUPV2_ENTRY_FORMAT, cgroup);
+		if ((EOF == rc) || (1 == rc)) {
```

Comment:
Won't the check for EOF stop it from looping to the next line on a unsuccessful match?

Reply:
Yes, EOF (end of file) will only occur once when the end of the `cgroupFile` is reached.

Reply:
I don't think we want to break on EOF; in a docker container I see many lines before the one starting with "0::", so this would stop before finding that line. The condition needs to be `1 == rc`.

Reply:
I tried a small test case and it doesn't stop on EOF, it reads all the lines. EOF is not returned, 0 is returned when the pattern doesn't match.
```
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv) {
        int rc;
        char buffer[1024];
        char cgroup[1024];
        FILE *cgroupFile = 0;
        printf("open %s\n", argv[1]);
        cgroupFile = fopen(argv[1], "r");
        if (NULL == cgroupFile) {
                perror("fopen");
                return -1;
        }
        while (0 == feof(cgroupFile)) {
                if (NULL == fgets(buffer, sizeof(buffer), cgroupFile)) {
                        perror("fgets");
                        return -1;
                }
                printf("%s", buffer);
                rc = sscanf(buffer, "0::%s", cgroup);
                printf("rc = %d EOF %d\n", rc, EOF);
                if ((EOF == rc) || (1 == rc)) break;
        }

        return 0;
}
```

Reply:
EOF should only occur once when all lines in the file are read. In the above condition, we are breaking out of the loop once `sscanf` returns EOF. `feof` will also exit the loop once EOF is encountered. Removed the EOF check here since it is redundant.